test/ui.spec.ts

```diff
@@ -94,279 +146,1444 @@ describe('Signer integration', function () {
+        timestamp: parsedApproveResult.timestamp,
+      });
+
+      expect(parsedApproveResult).to.deep.contain(expectedApproveResult);
+      expect(parsedApproveResult.id).to.equal(base58Encode(blake2b(bytes)));
+
+      expect(
+        verifySignature(issuer.publicKey, bytes, parsedApproveResult.proofs[0])
+      ).to.be.true;
+    });
   });
 
-  it('package tx', async function () {
-    await signedTxShouldBeValid.call(
-      this,
-      'batch',
-      [ISSUE, TRANSFER, REISSUE, BURN, LEASE, CANCEL_LEASE, ALIAS],
-      By.xpath("//div[contains(@class, 'package-transaction')]")
-    );
-    await signedTxShouldBeValid.call(
-      this,
-      'batch',
-      [MASS_TRANSFER, DATA, SET_SCRIPT, SPONSORSHIP, SET_ASSET_SCRIPT],
-      By.xpath("//div[contains(@class, 'package-transaction')]")
-    );
+  async function performDataTransaction(this: mocha.Context, data: DataArgs) {
+    const { waitForNewWindows } = await Windows.captureNewWindows.call(this);
+    await this.driver.executeScript(data => {
+      window.result = window.signer.data(data).broadcast({ confirmations: 0 });
+    }, data);
+    [messageWindow] = await waitForNewWindows(1);
+    await this.driver.switchTo().window(messageWindow);
+    await this.driver.navigate().refresh();
+  }
+
+  describe('Record in the account data storage', function () {
+    it('Write to Data storage', async function () {
+      const data: DataArgs = {
+        data: [
+          {
+            key: 'bool-entry',
+            value: false,
+            type: 'boolean',
+          },
+          {
+            key: 'str-entry',
+            value: 'Some string',
+            type: 'string',
+          },
+          {
+            key: 'binary',
+            value: 'base64:AbCdAbCdAbCdAbCdAbCdAbCdAbCdAbCdAbCdAbCdAbCd',
+            type: 'binary',
+          },
+          {
+            key: 'integer',
+            value: 20,
+            type: 'integer',
+          },
+        ],
+      };
+
+      await performDataTransaction.call(this, data);
+
+      await approveMessage.call(this);
+      await closeMessage.call(this);
+
+      const result = (await getApproveResult.call(this)) as [
+        BroadcastedTx<SignedTx<SignerDataTx>>
+      ];
+
+      const [parsedApproveResult] = result;
+      const expectedApproveResult = {
+        type: 12 as const,
+        version: 2,
+        senderPublicKey: issuer.publicKey,
+        data: data.data,
+        fee: 100000,
+        chainId,
+      };
+
+      const bytes = makeTxBytes({
+        ...expectedApproveResult,
+        timestamp: parsedApproveResult.timestamp,
+      });
+
+      expect(parsedApproveResult).to.deep.contain(expectedApproveResult);
+      expect(parsedApproveResult.id).to.equal(base58Encode(blake2b(bytes)));
+
+      expect(
+        verifySignature(issuer.publicKey, bytes, parsedApproveResult.proofs[0])
+      ).to.be.true;
+    });
+
+    it.skip('(unsupported) Remove entry from Data storage');
```

Comment:
May be add a note that this isn't supported because of validation in signer? Now it looks like our provider or keeper do not support it